src/styles/calendar.scss

```diff
@@ -0,0 +1,55 @@
+@import "./variables.scss";
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($main-size * 7) + ($gap * 6);
+  padding: 25px 10px 10px;
+  gap: $gap;
+
+  &__day {
+    box-sizing: border-box;
+    width: $main-size;
+    height: $main-size;
+
+    background-color: #eee;
+    border: 1px solid #000;
+
+    transition-duration: 0.5s;
+
+    &:hover {
+      background-color: pink;
+      cursor: pointer;
+
+      transition-duration: 0.5s;
+      transform: translateY(-20px);
+    }
+  }
+}
+
+@for $i from 1 through 31 {
+  .calendar__day:nth-child(#{$i}) {
+    text-align: center;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    &::before {
+      content: "#{$i}";
+      text-align: center;
+      line-height: $main-size;
+    }
+  }
+}
+
+$days: mon 0, tue 1, wed 2, thu 3, fri 4, sat 5, sun 6;
+
+@each $day, $number in $days {
+  .calendar--start-#{$day} .calendar__day:first-child {
+    margin-left: ($main-size + $gap) * $number;
+  }
+}
+
+@for $j from 28 through 31 {
+  .calendar--month-#{$j} :nth-child(n+#{$j + 1}) {
+    visibility: hidden;
```

Comment:
You should use `display:none` instead of `visibility:hidden` because your days exist in the calendar and interactive. Fix it